windows/src/desktop/kmshell/util/Keyman.System.KeymanStartTask.pas

```diff
@@ -247,10 +247,7 @@ class function TKeymanStartTask.GetTaskName: string;
 class procedure TKeymanStartTask.RecreateTask;
 begin
   if not Reg_GetDebugFlag(SRegValue_Flag_UseAutoStartTask, True) then
-  begin
-    DeleteTask;
     Exit;
-  end;
```

Comment:
Let's assume that the average user never had the task to start with, so we'll not waste time and risk errors connecting to the task scheduler service to try and delete it.

The task does no harm sitting there as it will not be fired. Uninstalling will remove it also.